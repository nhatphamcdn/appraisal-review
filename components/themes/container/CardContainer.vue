<!-- Script -->
<script setup lang="ts">
// Component interface
interface ICardCotnainerProps {
  boxShadow?: boolean;
  backgroundColor?: string;
  borderRadius?: number | string;
}

// Component define props
const props = withDefaults(defineProps<ICardCotnainerProps>(), {
  boxShadow: true,
  backgroundColor: "white",
  borderRadius: "0.5rem",
});

// Computed handle
const boxShadowStyleValue = computed(() => {
  const SHADOW_VALUE = "0 1px 3px 0 rgba(0,0,0,.1), 0 1px 2px 0 rgba(0,0,0,.06)";
  if (!props.boxShadow) {
    return "none";
  }
  return SHADOW_VALUE;
});

const borderRadiusStyleValue = computed(() => {
  if (typeof props.borderRadius === "number") {
    return `${props.borderRadius}px`;
  }
  return props.borderRadius;
});
</script>

<!-- DOM Render -->
<template>
  <div class="card-container">
    <slot />
  </div>
</template>

<!-- Scoped style -->
<style lang="scss" scoped>
.card-container {
  border-radius: v-bind(borderRadiusStyleValue);
  background-color: v-bind("props.backgroundColor");
  box-shadow: v-bind(boxShadowStyleValue);
}
</style>
