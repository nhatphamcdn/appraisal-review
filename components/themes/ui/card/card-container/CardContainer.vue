<!-- Script -->
<script setup lang="ts">
// Component interface
interface ICardCotnainerProps {
  boxShadow?: string;
  backgroundColor?: string;
  borderRadius?: string | number;
}

// Component define props
const props = withDefaults(defineProps<ICardCotnainerProps>(), {
  boxShadow: "0 1px 3px 0 rgba(0,0,0,.1), 0 1px 2px 0 rgba(0,0,0,.06)",
  backgroundColor: "white",
  borderRadius: "0.5rem",
});
</script>

<!-- DOM Render -->
<template>
  <div class="card-container">
    <slot />
  </div>
</template>

<!-- Scoped style -->
<style lang="scss" scoped>
.card-container {
  border-radius: v-bind("typeof props.borderRadius === 'number' ? `${props.borderRadius}px` : props.borderRadius");
  background-color: v-bind("props.backgroundColor");
  box-shadow: v-bind("props.boxShadow");
}
</style>
