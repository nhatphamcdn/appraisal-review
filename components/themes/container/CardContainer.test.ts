import { describe, expect, test } from "vitest";
import CardContainer from "./CardContainer.vue";
import { render } from "~/tests/testingLibraryVue";

describe("CardContainer", () => {
  test("mounts and is still the same", () => {
    const { html } = render(CardContainer, { props: { backgroundColor: "black" } });
    expect(html()).toMatchSnapshot();
  });
});
