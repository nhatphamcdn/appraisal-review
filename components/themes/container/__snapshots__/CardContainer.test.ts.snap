// Vitest Snapshot v1

exports[`CardContainer > mounts and is still the same 1`] = `"<div class=\\"card-container\\" data-v-5424779e=\\"\\"></div>"`;
